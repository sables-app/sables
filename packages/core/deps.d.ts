export {
  combineReducers,
  configureStore,
  createEntityAdapter,
  createSelector as createSelectorReduxToolkit,
  createSlice as createSliceReduxToolkit,
} from "@reduxjs/toolkit";
