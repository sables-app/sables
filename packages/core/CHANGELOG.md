# @sables/core

## 0.11.0

### Minor Changes

- 3cf28c4: Add `useActionCallback` hook.
- 3cf28c4: Rename `useWithDispatch` to `useAction`.

## 0.10.0

### Minor Changes

- cf721c6: Update `withProps` to accept a transform function.

## 0.9.1

### Patch Changes

- 4bfe027: Add missing reducer types to `EntitySlice`

## 0.9.0

### Minor Changes

- fd33a77: Expose entity adapter on entity slices.

## 0.7.1

### Patch Changes

- 9903795: Export `Selector` from `reselect`

## 0.7.0

### Minor Changes

- 31bdf2c: Add `distinctEntitySelectors` to assign distinctive names to entity adapter selectors. Update `createEntitySlice` to use `distinctEntitySelectors`.

## 0.6.0

### Minor Changes

- 4248daa: Export `Dictionary` from `@reduxjs/toolkit`.

## 0.5.2

### Patch Changes

- 8be70ce: Add `EntitySelectors` type to `EntitySlice`

## 0.5.1

### Patch Changes

- 66c78ca: Widen `Entity` generic to allow entity types to be defined using `interface`.

## 0.5.0

### Minor Changes

- ef12f71: Add `createEntitySlice` function

### Patch Changes

- 76f8a8c: Update `createSideEffect` documentation

## 0.4.0

### Minor Changes

- e93bc6e: Update `useSideEffect` to return an object

## 0.3.0

### Minor Changes

- 197bdac: Refactor `createSideEffect` to accept action creators

## 0.2.0

### Minor Changes

- 058d952: Rewrite `createSlice` to use a fluent interface

## 0.1.2

### Patch Changes

- 8863895: Move React to peer dependencies
- db2f705: Rename internal test export

## 0.1.1

### Patch Changes

- e5d4a08: Fix `Link` component `hash` handling
