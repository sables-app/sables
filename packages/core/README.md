# `@sables/core`

> The foundation of Sables, focusing on Redux-based state management.

Please view [the Core API documentation](https://sables.dev/docs/api#core) for more information.

## Install

```sh
npm install @sables/core
```
