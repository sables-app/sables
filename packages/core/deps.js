// import * as ReduxToolkit from "@reduxjs/toolkit/dist/redux-toolkit.modern.js";
// import * as ReduxToolkitMod from "@reduxjs/toolkit/dist/redux-toolkit.cjs.development.js";
import * as ReduxToolkitMod from "@reduxjs/toolkit/dist/redux-toolkit.cjs.production.min.js";

const ReduxToolkit = ReduxToolkitMod.default || ReduxToolkitMod;

const {
  combineReducers,
  configureStore,
  createEntityAdapter,
  createReducer,
  createSelector,
  createSlice,
} = ReduxToolkit;

export {
  combineReducers,
  configureStore,
  createEntityAdapter,
  createReducer,
  createSelector as createSelectorReduxToolkit,
  createSlice as createSliceReduxToolkit,
};
