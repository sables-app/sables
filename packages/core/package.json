{
  "name": "@sables/core",
  "version": "0.1.2",
  "description": "The foundation of Sables, focusing on Redux-based state management.",
  "author": "Morris Allison III (https://morris.xyz)",
  "license": "MIT",
  "bugs": "https://github.com/sables-app/sables/issues",
  "homepage": "https://sables.dev/docs/api#core",
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/sables-app/sables.git",
    "directory": "packages/core"
  },
  "type": "module",
  "module": "./dist/main.js",
  "main": "./dist/main.js",
  "dependencies": {
    "@reduxjs/toolkit": "^1.9.0",
    "@sables/utils": "~0.1.0",
    "nanoid": "^4.0.0",
    "react-redux": "^8.0.5",
    "redux": "^4.2.0",
    "reselect": "^4.1.7",
    "rxjs": "^7.8.0"
  },
  "peerDependencies": {
    "@types/react": "^18.0.0",
    "react": "^18.0.0"
  },
  "peerDependenciesMeta": {
    "@types/react": {
      "optional": true
    }
  },
  "devDependencies": {
    "@types/react": "^18.0.26",
    "react": "^18.2.0",
    "typescript": "^4.9.3"
  },
  "exports": {
    ".": "./dist/main.js",
    "./package.json": "./package.json",
    "./__internal__test": "./dist/__tests__/utils.js"
  },
  "files": [
    "dist",
    "deps.d.ts",
    "deps.js",
    "test.d.ts",
    "test.js",
    "!__tests__"
  ]
}
