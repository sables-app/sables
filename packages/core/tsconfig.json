{
  "extends": "@sables-app/scripts/tsconfig.package.json",
  "include": ["src/**/*", "src/**/*.json"],
  "exclude": ["__mocks__", "__snapshots__", "dist", "node_modules"],
  "compilerOptions": { "rootDir": "src", "outDir": "dist", "noEmit": false },
  "references": [{ "path": "../../packages/utils" }]
}
