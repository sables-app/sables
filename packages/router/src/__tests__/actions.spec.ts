import { hasLazyMeta } from "@sables/core";

import { describe, expect, it } from "vitest";

import {
  advanceLocation,
  chooseLocation,
  pushLocation,
  replaceLocation,
  retreatLocation,
} from "../actions.js";

describe("actions", () => {
  describe("chooseLocation", () => {
    it("creates a `go` action from redux-first-history", () => {
      expect(chooseLocation(1)).toMatchSnapshot();
    });

    it("is slice dependency", () => {
      expect(hasLazyMeta(chooseLocation)).toBe(true);
    });
  });

  describe("retreatLocation", () => {
    it("creates a `goBack` action from redux-first-history", () => {
      expect(retreatLocation()).toMatchSnapshot();
    });

    it("is slice dependency", () => {
      expect(hasLazyMeta(retreatLocation)).toBe(true);
    });
  });

  describe("advanceLocation", () => {
    it("creates a `goForward` action from redux-first-history", () => {
      expect(advanceLocation()).toMatchSnapshot();
    });

    it("is slice dependency", () => {
      expect(hasLazyMeta(advanceLocation)).toBe(true);
    });
  });

  describe("pushLocation", () => {
    it("creates a `push` action from redux-first-history", () => {
      expect(pushLocation("/")).toMatchSnapshot();
    });

    it("is slice dependency", () => {
      expect(hasLazyMeta(pushLocation)).toBe(true);
    });
  });

  describe("replaceLocation", () => {
    it("creates a `replace` action from redux-first-history", () => {
      expect(replaceLocation("/")).toMatchSnapshot();
    });

    it("is slice dependency", () => {
      expect(hasLazyMeta(replaceLocation)).toBe(true);
    });
  });
});
