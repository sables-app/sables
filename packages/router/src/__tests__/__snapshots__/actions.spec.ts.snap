// Vitest Snapshot v1

exports[`actions > advanceLocation > creates a \`goForward\` action from redux-first-history 1`] = `
{
  "error": false,
  "meta": {
    Symbol(@sables/lazyMeta): {
      "observers": Set {},
      "slices": Set {},
    },
  },
  "payload": {
    "args": [],
    "method": "goForward",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`actions > chooseLocation > creates a \`go\` action from redux-first-history 1`] = `
{
  "error": false,
  "meta": {
    Symbol(@sables/lazyMeta): {
      "observers": Set {},
      "slices": Set {},
    },
  },
  "payload": {
    "args": [
      1,
    ],
    "method": "go",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`actions > pushLocation > creates a \`push\` action from redux-first-history 1`] = `
{
  "error": false,
  "meta": {
    Symbol(@sables/lazyMeta): {
      "observers": Set {},
      "slices": Set {},
    },
  },
  "payload": {
    "args": [
      "/",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`actions > replaceLocation > creates a \`replace\` action from redux-first-history 1`] = `
{
  "error": false,
  "meta": {
    Symbol(@sables/lazyMeta): {
      "observers": Set {},
      "slices": Set {},
    },
  },
  "payload": {
    "args": [
      "/",
    ],
    "method": "replace",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`actions > retreatLocation > creates a \`goBack\` action from redux-first-history 1`] = `
{
  "error": false,
  "meta": {
    Symbol(@sables/lazyMeta): {
      "observers": Set {},
      "slices": Set {},
    },
  },
  "payload": {
    "args": [],
    "method": "goBack",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;
